<launch>

  <arg name="robot_model"                       default=""/>
  <arg name="robot_name"                        default="$(arg robot_model)"/>
  <arg name="use_rviz"                          default="true"/>
  <arg name="mode_configs"                      default="$(find interbotix_xshexapod_joy)/config/modes.yaml"/>
  <arg name="load_configs"                      default="true"/>
  <arg name="use_pixels"                        default="true"/>
  <arg name="threshold"                         default="0.5"/>
  <arg name="controller"                        default="ps4"/>
  <arg name="x_stride"                          default="0.06"/>
  <arg name="y_stride"                          default="0.06"/>
  <arg name="yaw_stride"                        default="0.2618"/>
  <arg name="x_pos_yaw_fudge"                   default="0"/>
  <arg name="x_neg_yaw_fudge"                   default="0"/>
  <arg name="y_pos_yaw_fudge"                   default="0"/>
  <arg name="y_neg_yaw_fudge"                   default="0"/>
  <arg name="p_gain"                            default="800"/>
  <arg name="move_step_time"                    default="0.15"/>
  <arg name="num_steps"                         default="20"/>
  <arg name="max_foot_height"                   default="0.04"/>
  <arg name="launch_driver"                     default="true"/>

  <include if="$(arg launch_driver)" file="$(find interbotix_xshexapod_control)/launch/xshexapod_control.launch">
    <arg name="robot_model"                       value="$(arg robot_model)"/>
    <arg name="robot_name"                        value="$(arg robot_name)"/>
    <arg name="use_rviz"                          value="$(arg use_rviz)"/>
    <arg name="rviz_frame"                        value="$(arg robot_name)/odom"/>
    <arg name="mode_configs"                      value="$(arg mode_configs)"/>
    <arg name="load_configs"                      value="$(arg load_configs)"/>
    <arg name="use_pixels"                        value="$(arg use_pixels)"/>
  </include>

  <node
    pkg="joy"
    type="joy_node"
    name="joy_node"
    output="screen"
    ns="$(arg robot_name)">
    <param name="dev"                             value="/dev/input/js0" />
    <remap from="joy"                             to="commands/joy_raw"/>
  </node>

  <node
    pkg="interbotix_xshexapod_joy"
    type="xshexapod_joy"
    name="xshexapod_joy"
    output="screen"
    ns="$(arg robot_name)">
    <param name="threshold"                       value="$(arg threshold)"/>
    <param name="controller"                      value="$(arg controller)"/>
  </node>

  <node
    pkg="interbotix_xshexapod_joy"
    type="xshexapod_robot"
    name="xshexapod_robot"
    output="screen"
    ns="$(arg robot_name)">
    <param name="x_stride"                        value="$(arg x_stride)"/>
    <param name="y_stride"                        value="$(arg y_stride)"/>
    <param name="yaw_stride"                      value="$(arg yaw_stride)"/>
    <param name="x_pos_yaw_fudge"                 value="$(arg x_pos_yaw_fudge)"/>
    <param name="x_neg_yaw_fudge"                 value="$(arg x_neg_yaw_fudge)"/>
    <param name="y_pos_yaw_fudge"                 value="$(arg y_pos_yaw_fudge)"/>
    <param name="y_neg_yaw_fudge"                 value="$(arg y_neg_yaw_fudge)"/>
    <param name="p_gain"                          value="$(arg p_gain)"/>
    <param name="move_step_time"                  value="$(arg move_step_time)"/>
    <param name="num_steps"                       value="$(arg num_steps)"/>
    <param name="max_foot_height"                 value="$(arg max_foot_height)"/>
    <param name="robot_model"                     value="$(arg robot_model)"/>
  </node>

</launch>
